compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUCheckIRBeforeLLVMConversion.cpp

```diff
@@ -58,11 +59,14 @@ checkStackAllocationSize(mlir::FunctionOpInterface funcOp) {
       allocaSize *= dimSize;
     }
     for (auto operand : allocaOp.getDynamicSizes()) {
-      auto ub = ValueBoundsConstraintSet::computeConstantBound(
-          presburger::BoundType::UB, operand, /*dim=*/std::nullopt,
-          /*stopCondition=*/nullptr, /*closedUB=*/true);
-      if (succeeded(ub)) {
-        allocaSize *= ub.value();
+      // Assume vscale is 1 for determining if the alloca is within the stack
+      // limit. This should always resolve to a constant bound.
+      auto ub = vector::ScalableValueBoundsConstraintSet::computeScalableBound(
+          operand, /*dim=*/std::nullopt,
+          /*vscaleMin=*/1, /*vscaleMax=*/1, presburger::BoundType::UB);
+      FailureOr<vector::ConstantOrScalableBound::BoundSize> size;
+      if (succeeded(ub) && succeeded(size = ub->getSize())) {
+        allocaSize *= size->baseSize;
```

Comment:
I don't get why we need `size`. Can't this just be below?

```
      if (succeeded(ub)) {
        allocaSize *= ub->getSize()->baseSize;
        continue;
      }
```

Reply:
Probably overly cautious,  `getSize()` can fail if the bound is a more complex expression (in terms of vscale). However, that should never happen if vscale is a constant (which it is here). So `ub->getSize()->baseSize` should be fine.